src/main/java/com/sublinks/sublinksapi/comment/services/CommentService.java

```diff
@@ -96,6 +100,25 @@ public void createComment(final Comment comment, final Comment parent) {
     createComment(comment);
   }
 
+  /**
+   * Deletes a comment and publishes an event.
+   *
+   * @param comment The Comment object to be deleted.
+   */
+  public Comment deleteComment(final Comment comment) {
+    String DELETED_REPLACEMENT_TEXT = "*Permanently Deleted*";
+
+    Comment foundComment = commentRepository.findById(comment.getId())
+        .orElseThrow(() -> {
+          logger.error("no comment found by id: {}", comment.getId());
+          return new EntityNotFoundException("could not find comment by Id");
+        });
+    foundComment.setCommentBody(DELETED_REPLACEMENT_TEXT);
+    foundComment.setRemovedState(RemovedState.REMOVED);
```

Comment:
Best you create a new RemovedState, something like "PURGED" so it can be identified that it was purged and not just removed or some weird stuff.

Reply:
Okay sounds good - I was trying to look at the lemmy-ui to see if there was an expected state specifically for purged but I'm not super familiar with that code.